tests/ssl-opt.sh

```diff
@@ -4608,6 +4641,7 @@ run_test    "Authentication, CA callback: client max_int+1 chain, server require
             -s "use CA callback for X.509 CRT verification" \
             -s "X509 - A fatal error occurred"
 
+requires_config_value_equals "MBEDTLS_X509_MAX_INTERMEDIATE_CA" $MAX_IM_CA
 requires_full_size_output_buffer
 requires_config_enabled MBEDTLS_X509_TRUSTED_CERTIFICATE_CALLBACK
 run_test    "Authentication, CA callback: client max_int chain, server required" \
```

Comment:
Missing `requires_max_content_len 4096` on the test case `"SSLv3 with extensions, server side"`.

I found this by checking that each use of `max_frag_len` has a corresponding `requires_max_content_len` (or doesn't need one). This check needs to be done for each Mbed TLS versions since different versions have different test cases. Here there's an extra test case in 2.2x and 2.16 for a feature that was removed in 3.0.